pkg/controller/namespace/deletion/namespaced_resources_deleter.go

```diff
@@ -563,3 +567,69 @@ func (d *namespacedResourcesDeleter) estimateGracefulTerminationForPods(ns strin
 	}
 	return estimate, nil
 }
+
+// updateStatusCondition does best effort to update status condition for
+// a namespace if namespace deletion didn't finish. In case of a failure, it
+// does not propagate its error further, instead it logs it and the error
+// from namespace deletion is propagated.
+func (d *namespacedResourcesDeleter) updateStatusCondition(nsName string, deletionErr error) {
+	err := retry.RetryOnConflict(retry.DefaultBackoff, func() error {
```

Comment:
I recommend not retrying--if it fails, just wait until the whole thing gets called again?